File: app/src/main/java/com/marked/pixsee/signup/SignUpPassFragment.java
package com.marked.pixsee.signup;

import android.content.Context;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.AutoCompleteTextView;
import android.widget.Button;
import android.widget.Toast;

import com.marked.pixsee.R;


public class SignUpPassFragment extends Fragment {
	private Button nextButton;
	private AutoCompleteTextView passwordEditText;

	public SignUpPassFragment() {
	}

	private SignUpPassFragmentInteraction mListener;

	@Nullable
	@Override
	public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
		View rootView = inflater.inflate(R.layout.fragment_sign_up_pass, container, false);
		nextButton = (Button) rootView.findViewById(R.id.nextButton);
		passwordEditText = (AutoCompleteTextView) rootView.findViewById(R.id.passwordEditText);
		nextButton.setOnClickListener(new View.OnClickListener() {
			@Override
			public void onClick(View v) {
				String password = passwordEditText.getText().toString().trim();
				if (password.isEmpty())
					Toast.makeText(getContext(), "Please enter a password", Toast.LENGTH_SHORT).show();
				else if (password.length() < 6)
					Toast.makeText(getContext(), "The password must be at least 6 characters long", Toast.LENGTH_SHORT).show();
				else
					onNextPressed(password);
			}
		});

		// Inflate the layout for this fragment
		return rootView;
	}

	// TODO: Rename method, update argument and hook method into UI event
	void onNextPressed(String password) {
		if (mListener != null) {
			mListener.onSavePassword(password);
		}
	}

	@Override
	public void onAttach(Context context) {
		super.onAttach(context);
		if (context instanceof SignUpPassFragmentInteraction) {
			mListener = (SignUpPassFragmentInteraction) context;
		} else throw new RuntimeException(context.toString() + " must implement SignUpEmailFragmentInteraction");
	}


	interface SignUpPassFragmentInteraction {
		// TODO: Update argument type and name
		void onSavePassword(String password);
	}


	public static SignUpPassFragment newInstance() {
		return new SignUpPassFragment();
	}
}// Required empty public constructor

File: app/src/main/java/com/marked/pixsee/fragment/signup/SignUpPassFragment.kt
package com.marked.pixsee.fragment.signup

import android.content.Context
import android.os.Bundle
import android.support.v4.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.marked.pixsee.R
import kotlinx.android.synthetic.main.activity_log_in.*
import kotlinx.android.synthetic.main.fragment_sign_up_pass.view.*
import org.jetbrains.anko.onClick
import org.jetbrains.anko.toast

class SignUpPassFragment : Fragment() {
	private var mListener: SignUpPassFragmentInteraction? = null

	override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
		val rootView = inflater.inflate(R.layout.fragment_sign_up_pass, container, false)
		rootView.nextButton.onClick {
			val password = passwordEditText.text.toString().trim { it <= ' ' }
			if (password.isNullOrBlank())
				activity.toast("Please enter a password")
			else if (password.length < 6)
				activity.toast("The password must be at least 6 characters long")
			else
				onNextPressed(password)
		}
		// Inflate the layout for this fragment
		return rootView
	}

	// TODO: Rename method, update argument and hook method into UI event
	fun onNextPressed(password: String) {
		if (mListener != null) {
			mListener?.onSavePassword(password)
		}
	}

	override fun onAttach(context: Context?) {
		super.onAttach(context)
		if (context is SignUpPassFragmentInteraction) {
			mListener = context as SignUpPassFragmentInteraction?
		} else throw RuntimeException(context.toString() + " must implement SignUpEmailFragmentInteraction")
	}

	override fun onDetach() {
		super.onDetach()
		mListener = null
	}


	interface SignUpPassFragmentInteraction {
		// TODO: Update argument type and name
		fun onSavePassword(password: String)
	}

	companion object {
		fun newInstance(): SignUpPassFragment {
			return SignUpPassFragment()
		}
	}
}// Required empty public constructor
